django/db/backends/base/schema.py

```diff
@@ -1063,9 +1063,21 @@ def _field_should_be_altered(self, old_field, new_field):
         _, new_path, new_args, new_kwargs = new_field.deconstruct()
         # Don't alter when:
         # - changing only a field name
+        # - changing an attribute that doesn't affect the schema
         # - adding only a db_column and the column name is not changed
-        old_kwargs.pop('db_column', None)
-        new_kwargs.pop('db_column', None)
+        non_database_fields = [
+            'blank',
+            'db_column',  # only if the column name is not changed
+            'editable',
+            'error_messages',
+            'help_text',
+            'on_delete',
+            'related_name',
+            'verbose_name',
+        ]
```

Comment:
I think we missed some attributes e.g. `related_query_name`, `limit_choices_to`, or `validators`.

Reply:
I added those three, what else do you think we are missing?

Reply:
I guess `name` would be another one.